### PR TITLE
TST: groupby apply with indexing and column aggregation returns the column #7002

### LINKED ISSUE #7002: BUG/TST: verify that groupby apply with a column aggregation does not return the column
related https://github.com/pydata/pandas/pull/7000

```
In [1]:  df = DataFrame({'foo1' : ['one', 'two', 'two', 'three', 'one', 'two'],
                                      'foo2' : np.random.randn(6)})

In [2]: df
Out[2]: 
    foo1      foo2
0    one  1.006666
1    two  0.002063
2    two  1.507785
3  three  1.865921
4    one  0.141202
5    two -1.079792

[6 rows x 2 columns]
```

```
In [3]: df.groupby('foo1').mean()
Out[3]: 
           foo2
foo1           
one    0.573934
three  1.865921
two    0.143352

[3 rows x 1 columns]

In [4]: df.groupby('foo1').apply(lambda x: x.mean())
Out[4]: 
           foo2
foo1           
one    0.573934
three  1.865921
two    0.143352

[3 rows x 1 columns]
```

This should return the foo1 column as well

```
[6]: df.groupby('foo1',as_index=False).apply(lambda x: x.mean())
Out[6]: 
       foo2
0  0.573934
1  1.865921
2  0.143352

[3 rows x 1 columns]

In [7]: df.groupby('foo1',as_index=False).mean()
Out[7]: 
    foo1      foo2
0    one  0.573934
1  three  1.865921
2    two  0.143352

[3 rows x 2 columns]

```

### PR DESCRIPTION
**Pull request by fangchenli**
Adds test for #7002 

- [x] closes #7002 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/groupby/test_apply.py [hunk 1]
```diff
@@ -961,3 +961,17 @@ def fn(x):
...
+def test_apply_function_with_indexing_return_column():
+    # GH: 7002
+    df = DataFrame(
+        {
+            "foo1": ["one", "two", "two", "three", "one", "two"],
+            "foo2": np.random.randn(6),
+        }
+    )
+    result = df.groupby("foo1", as_index=False).apply(lambda x: x.mean())
+    expected = df.groupby("foo1", as_index=False).mean()
```
Comment (dsaxton) [inline pandas/tests/groupby/test_apply.py:975]: Can you explicitly construct the expected DataFrame here? Makes it easier to understand what's being tested and to constrain testing to a single implementation (groupby.apply). (Might be helpful to work with simple, deterministic test data for that.)

File: pandas/tests/groupby/test_apply.py [hunk 2]
```diff
@@ -961,3 +961,17 @@ def fn(x):
...
+    tm.assert_frame_equal(result, expected)
+    assert "foo1" in result.columns
```
Comment (dsaxton) [inline pandas/tests/groupby/test_apply.py:977]: You shouldn't need this extra assertion

File: pandas/tests/groupby/test_apply.py (hunk 1 above)
Comment (VirosaLi) [reply pandas/tests/groupby/test_apply.py]: I've fixed everything you suggested.

File: pandas/tests/groupby/test_apply.py [hunk 3]
```diff
@@ -155,6 +155,7 @@ def test_group_apply_once_per_group(df, group_names):
     # once per group
 
     names = []
+
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_apply.py:158]: can you revert this change

File: pandas/tests/groupby/test_apply.py [hunk 4]
```diff
@@ -683,7 +684,6 @@ def func_with_date(batch):
 
 
 def test_gb_apply_list_of_unequal_len_arrays():
-
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_apply.py:686]: can you revert this change

**Review by simonjayhawkins — APPROVED** [review]
Thanks @VirosaLi generally lgtm. a couple of unrelated changes crept in. can you revert those.

**Comment by WillAyd** [comment]
@VirosaLi can you address @simonjayhawkins comments? Lgtm otherwise

**Comment by simonjayhawkins** [comment]
@VirosaLi Thanks for the updates. lgtm. ping on green

**Comment by VirosaLi** [comment]
One of the checks is on pending for hours. Is this normal?

**Comment by dsaxton** [comment]
> One of the checks is on pending for hours. Is this normal?

Travis is glitchy sometimes, it's green if you click through

**Comment by jreback** [comment]
thanks @VirosaLi
